bindings/java/src/main/com/apple/foundationdb/MappedKeyValueV2.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * MappedKeyValueV2.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb;
+
+import com.apple.foundationdb.tuple.ByteArrayUtil;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class MappedKeyValueV2 extends KeyValue {
+	private final byte[] rangeBegin;
+	private final byte[] rangeEnd;
+	private final byte[] responseBytes;
+	private final List<KeyValue> rangeResult;
+
+	// now it has 4 fields, key, value, getRange.begin, getRange.end
+	// this needs to change if FDBMappedKeyValueV2 definition is changed.
+	private static final int TOTAL_SERIALIZED_FIELD_FDBMappedKeyValueV2 = 5;
+
+	public MappedKeyValueV2(byte[] key, byte[] value, byte[] rangeBegin, byte[] rangeEnd, List<KeyValue> rangeResult,
+	                        byte[] responseBytes) {
+		super(key, value);
+		this.rangeBegin = rangeBegin;
+		this.rangeEnd = rangeEnd;
+		this.rangeResult = rangeResult;
+		this.responseBytes = responseBytes;
+	}
+
+	public byte[] getRangeBegin() { return rangeBegin; }
+
+	public byte[] getRangeEnd() { return rangeEnd; }
+
+	public List<KeyValue> getRangeResult() { return rangeResult; }
+
+	public byte[] getResponseBytes() { return responseBytes; }
```

Comment:
Is there a more ergonomic way of returning the "response bytes" here? At present, I think it's not clear to the user what they are supposed to do with this

Reply:
yeah you are correct, will change.